kirby/web/endpoints.py

```diff
@@ -1,37 +1,59 @@
-import sqlalchemy
+from sqlalchemy import func
+from sqlalchemy.orm.exc import NoResultFound
 from flask import abort
+from datetime import datetime
 from flask_restplus import Api, Resource, reqparse
-from ..models import db, Job, JobType
+from ..models import db, Script, Topic
 
 api = Api()
 
 registration_parser = reqparse.RequestParser()
-registration_parser.add_argument("name", type=str, required=True)
-registration_parser.add_argument("description", type=str)
-registration_parser.add_argument(
-    "type", choices=["scheduled", "triggered"], required=True
-)
+registration_parser.add_argument("script_id", type=int)
+registration_parser.add_argument("source_id", type=int, action="append")
+registration_parser.add_argument("destination_id", type=int, action="append")
 
 
 @api.route("/registration")
 class Registration(Resource):
     @api.expect(registration_parser)
-    def post(self):
+    def patch(self):
         args = registration_parser.parse_args()
+
         try:
-            db.session.add(
-                Job(
-                    name=args["name"],
-                    description=args.get("description"),
-                    type=JobType(args["type"]),
-                )
+            # Update last_seen
+            script = (
+                db.session.query(Script).filter_by(id=args["script_id"]).one()
             )
+            script.last_seen = datetime.utcnow()
+
+            # Add sources and destinations
+            for source_id in args["source_id"]:
+                source = db.session.query(Topic).filter_by(id=source_id).one()
+                script.add_source(source)
+            for destination_id in args["destination_id"]:
+                destination = (
+                    db.session.query(Topic).filter_by(id=destination_id).one()
+                )
+                script.add_destination(destination)
+
             db.session.commit()
-            return {
-                "id": db.session.query(Job)
-                .filter_by(name=args["name"])
-                .one()
-                .id
-            }
-        except sqlalchemy.exc.IntegrityError:
-            abort(400, "There is already a Job with this name.")
+
+            return {"message": "Sucess"}
+        except NoResultFound:
```

Comment:
Nice, but a bit lazy ;-)

Ideally you want to have `try..except` blocks as narrow as possible, so your exceptions are properly qualified.

Please use one try..except block per query, and provide meaningful error message

Reply:
They are now separated. Also session rollback if any query fails